scripts/azureml-assets/azureml/assets/model/model_utils.py

```diff
@@ -0,0 +1,173 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Model utils Operations Class."""
+
+import requests
+import os
+import azureml.assets as assets
+from azure.identity import DefaultAzureCredential
+from azure.ai.ml import load_model
+from azureml.assets.util import logger
+from azureml.assets.config import PathType
+from azureml.assets.model.download_utils import copy_azure_artifacts, download_git_model
+from pathlib import Path
+
+
+class RegistryUtils:
+    """Registry utils."""
+
+    INTERNAL_REGION_ASSET_STORE_URL = "https://int.api.azureml-test.ms/assetstore/v1.0"
+    PROD_REGION_ASSET_STORE_URL = "https://eastus.api.azureml.ms/assetstore/v1.0"
+    TOKEN_CREDENTIAL_URL = "https://management.azure.com/.default"
+    CREATE_DATA_REFERENCE_METHOD = "/temporaryDataReference/createOrGet"
+    DEV_REGISTRY_NAME = "azureml-dev"
+    RETRY_COUNT = 3
+
+    def get_assetstore_url(registry_name: str):
+        """Return asset URI for the registry."""
+        # USE https://eastus.api.azureml.ms/registrymanagement/v1.0/registries/{registryName}/discovery
+        if RegistryUtils.DEV_REGISTRY_NAME == registry_name:
+            return RegistryUtils.INTERNAL_REGION_ASSET_STORE_URL
+        return RegistryUtils.PROD_REGION_ASSET_STORE_URL
+
+    def get_registry_data_reference(registry_name: str, model_name: str, model_version: str):
```

Comment:
This works as a proof-of-concept, but we really should leverage the SDK as much as possible, which will allow this to support sovereign and air-gapped regions after model storage accounts are created there. You'd need to create an MLClient ([example](https://msdata.visualstudio.com/Vienna/_git/azureml-asset?path=/msftkube/scripts/create_environments.py)), which then gives you most of what you'd need to call [get_sas_uri_for_registry_asset](https://github.com/Azure/azure-sdk-for-python/blob/3c9b59c72f6229bd3986c92315b388f920ccc0c1/sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_registry_utils.py#L81). This is how the SDK calls it from [model create/update](https://github.com/Azure/azure-sdk-for-python/blob/3c9b59c72f6229bd3986c92315b388f920ccc0c1/sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py#L171).

Reply:
I'm a little leery about introducing azure-ai-ml dependency, but maybe this just needs to happen at some point anyway.